app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -91,6 +100,16 @@ class BrowserFragment : WebFragment(), CursorEvent {
         })
     }
 
+    override fun onEvent(event: BrowserNavigationOverlay.NavigationEvent, value: String?, autocompleteResult: InlineAutocompleteEditText.AutocompleteResult?) {
```

Comment:
I would prefer if this was set on a separate class with a BrowserFragment reference in order to separate responsibilities. fwiw, this is how swift conventionally does it (with the delegates as an extension at the bottom of the file). e.g.:
```kotlin
class BrowserNavigationOverlay() {
    ...
}

class FragmentEventHandler(private val browserFragment: BrowserFragment) {
    override onEvent ...
}
```

Then again, maybe it's awkward to keep a reference to BrowserFragment if you pass in a lambda like I suggested earlier... I'm not sure! Sorry I don't have all the best answers, just a lot of possibilities... :\